src/directory_patcher.rs

```diff
@@ -60,11 +60,26 @@ impl DirectoryPatcher {
     fn build_walker(&self) -> ignore::Walk {
         let mut types_builder = ignore::types::TypesBuilder::new();
         types_builder.add_defaults();
+        let mut cnt: u32 = 0;
         for t in &self.settings.selected_file_types {
-            types_builder.select(t);
+            if t.contains("*") {
+                let new_type = format!("type{}", cnt);
+                types_builder.add(&new_type, t).unwrap();
```

Comment:
`ignore::types::TypesBuilder::add` only returns error  when the `name` is wrong, not the `glob`. We control the `name` in this case so it can't go wrong. Thus I used `unwrap` here.

Reply:
Good point - consider putting that as a comment so that the next person to read the code know about this :)